src/core-services/tags/queries/tag.js

```diff
@@ -17,36 +17,22 @@ export default async function tag(context, input) {
   const { Tags } = collections;
   const { slugOrId, shopId, shouldIncludeInvisible = false } = input;
 
-  // Check to make sure user has `read` permissions for this tag
-  // TODO(auth-pod): revisit this check once legacyRoles are removed
-  // await context.validatePermissions("reaction:legacy:tags:${slugOrId}", "read", {
-  //   shopId,
-  //   legacyRoles: ["admin", "owner", "tags", "any"]
-  // });
-
-  // Check to see if user has `read` permissions for hidden / deleted tags
-  // TODO(pod-auth): revisit using `inactive` in resource, and revisit the word `inactive`
-  const hasInactivePermissions = await context.userHasPermission(`reaction:legacy:tags-inactive:${slugOrId}`, "read", {
+  const foundTag = await Tags.findOne({
+    $or: [{ _id: slugOrId }, { slug: slugOrId }],
     shopId
   });
 
-  let query = {
-    $and: [
-      { isVisible: true },
-      { shopId },
-      { $or: [{ _id: slugOrId }, { slug: slugOrId }] }
-    ]
-  };
+  // Check to see if user has `read` permissions for invisible tags
+  const hasInactivePermissions = await context.userHasPermission(`reaction:legacy:tags:${tag._id}`, "read:invisible", {
```

Comment:
Should be `${foundTag._id}` and move the `if (!foundTag)` check above this in case it isn't found